File: java-client-sdk/lib/src/main/java/org/openapis/openapi/models/shared/ProjectStatusResponsePayload.java
package org.openapis.openapi.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import java.time.OffsetDateTime;
import org.openapis.openapi.utils.DateTimeDeserializer;
import org.openapis.openapi.utils.DateTimeSerializer;

/**
 * ProjectStatusResponsePayload
 * Response format returned by the getRunStatus endpoint
**/
public class ProjectStatusResponsePayload {
    @JsonProperty("elapsedTime")
    public Double elapsedTime;
    public ProjectStatusResponsePayload withElapsedTime(Double elapsedTime) {
        this.elapsedTime = elapsedTime;
        return this;
    }
    
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("endTime")
    public OffsetDateTime endTime;
    public ProjectStatusResponsePayload withEndTime(OffsetDateTime endTime) {
        this.endTime = endTime;
        return this;
    }
    
    @JsonProperty("projectId")
    public String projectId;
    public ProjectStatusResponsePayload withProjectId(String projectId) {
        this.projectId = projectId;
        return this;
    }
    
    @JsonProperty("projectVersion")
    public Integer projectVersion;
    public ProjectStatusResponsePayload withProjectVersion(Integer projectVersion) {
        this.projectVersion = projectVersion;
        return this;
    }
    
    @JsonProperty("runId")
    public String runId;
    public ProjectStatusResponsePayload withRunId(String runId) {
        this.runId = runId;
        return this;
    }
    
    @JsonProperty("runUrl")
    public String runUrl;
    public ProjectStatusResponsePayload withRunUrl(String runUrl) {
        this.runUrl = runUrl;
        return this;
    }
    
    @JsonSerialize(using = DateTimeSerializer.class)
    @JsonDeserialize(using = DateTimeDeserializer.class)
    @JsonProperty("startTime")
    public OffsetDateTime startTime;
    public ProjectStatusResponsePayload withStartTime(OffsetDateTime startTime) {
        this.startTime = startTime;
        return this;
    }
    
    @JsonProperty("status")
    public ProjectRunStatusEnum status;
    public ProjectStatusResponsePayload withStatus(ProjectRunStatusEnum status) {
        this.status = status;
        return this;
    }
    
    @JsonProperty("traceId")
    public String traceId;
    public ProjectStatusResponsePayload withTraceId(String traceId) {
        this.traceId = traceId;
        return this;
    }
    
}

File: python-client-sdk/src/sdk/models/shared/projectstatusresponsepayload.py
from __future__ import annotations
import dataclasses
import dateutil.parser
from ..shared import projectrunstatus_enum as shared_projectrunstatus_enum
from dataclasses_json import Undefined, dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils


@dataclass_json(undefined=Undefined.EXCLUDE)
@dataclasses.dataclass
class ProjectStatusResponsePayload:
    r"""ProjectStatusResponsePayload
    Response format returned by the getRunStatus endpoint
    """
    
    elapsed_time: float = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('elapsedTime') }})
    end_time: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('endTime'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    project_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('projectId') }})
    project_version: int = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('projectVersion') }})
    run_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('runId') }})
    run_url: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('runUrl') }})
    start_time: datetime = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('startTime'), 'encoder': utils.datetimeisoformat(False), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    status: shared_projectrunstatus_enum.ProjectRunStatusEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    trace_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('traceId') }})
    